src/GraphQL.Tests/Types/ComplexGraphTypeTests.cs

```diff
@@ -122,6 +122,24 @@ public void create_field_with_func_resolver()
             type.Fields.Last().Type.ShouldBe(typeof(StringGraphType));
         }
 
+        [Fact]
+        public void throws_informative_exception_when_no_types_defined()
+        {
+            var type = new ComplexType<Droid>();
+
+            var fieldType = new FieldType()
+            {
+                Name = "name",
+                ResolvedType = null,
+                Type = null,
+            };
+
+            var exception = Should.Throw<ArgumentOutOfRangeException>(() => type.AddField(fieldType));
+
+            exception.ParamName.ShouldBe("Type");
+            exception.Message.ShouldStartWith("FieldType.Type is required when ResolvedType is unspecified.");
```

Comment:
Little bit of clarity on the request to add names.  I was thinking the name of the field  and the GraphType vs. the name of the `FieldType` class.  With the intent being it would be easier to track down which class has the error.

Some potential examples:

> The declared field 'name' on 'MyGraphQL.MyGraphType' requires a field 'Type' when no 'ResolvedType' is provided.

> The declared field 'name' on 'ComplexType\<Droid\>' requires a field 'Type' when no 'ResolvedType' is provided.

Reply:
Link for extension method for a "friendly type name".

https://stackoverflow.com/a/26429045/279764

```
    public static string GetFriendlyName(this Type type)
    {
        string friendlyName = type.Name;
        if (type.IsGenericType)
        {
            int iBacktick = friendlyName.IndexOf('`');
            if (iBacktick > 0)
            {
                friendlyName = friendlyName.Remove(iBacktick);
            }
            friendlyName += "<";
            Type[] typeParameters = type.GetGenericArguments();
            for (int i = 0; i < typeParameters.Length; ++i)
            {
                string typeParamName = GetFriendlyName(typeParameters[i]);
                friendlyName += (i == 0 ? typeParamName : "," + typeParamName);
            }
            friendlyName += ">";
        }

        return friendlyName;
    }
```